geopandas/geodataframe.py

```diff
@@ -445,14 +444,36 @@ def crs(self):
     @crs.setter
     def crs(self, value):
         """Sets the value of the crs"""
-        if self._geometry_column_name not in self:
+        if self._geometry_column_name is None:
+            raise ValueError(
+                "Assigning CRS to a GeoDataFrame without a geometry column is not "
+                "supported. Use GeoDataFrame.set_geometry to set the active "
+                "geometry column.",
```

Comment:
Can we first define the two variants of messages, and then reuse them below? (if I am correctly reading, there are two variants that are currently both used twice?)

Reply:
whoops sorry, this is a mess right now, I re-wrote this to have less branches and apparently commited part of my WIP changes as well.